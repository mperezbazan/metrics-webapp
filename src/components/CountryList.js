import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { getWeatherAPI } from '../redux/Country/Country';
import Country from './Country';
import Loading from './Loading';

const CountryList = () => {
  const [mainCountry, setMainCountry] = useState(Math.floor(Math.random() * 243));
  const [filteredCountries, setFilteredCountries] = useState([]);
  const countries = useSelector((state) => state.countries);
  useEffect(() => {
    setMainCountry(Math.floor(Math.random() * 243));
    getWeatherAPI();
  }, []);
  useEffect(() => {
    setFilteredCountries(countries);
  }, [countries]);
  const handleChange = (e) => {
    const search = e.target.value;
    const newCountries = [...countries];
    const data = newCountries.filter((country) => (
      country.country.toLowerCase().includes(search.toLowerCase())));
    setFilteredCountries(data);
  };
  return (
    <>
      <div>
        {
        countries.length === 0
          ? <Loading />
          : (
            <>
              <Country item={countries[mainCountry]} fullwidth color main />
              <div className="home-mid-container">
                <p className="home-title">WEATHER BY COUNTRY</p>
                <input type="text" onChange={handleChange} className="search-input" placeholder="Search by country" />
              </div>
              <div className="countries-container">
                {
                  filteredCountries.map((country, index) => {
                    let background = false;
                    if ((index + 1) % 4 === 0 || (index + 4) % 4 === 0 || index === 0) {
                      background = true;
                    }
                    return <Country key={country.alpha3} item={country} background={background} />;
                  })
                }
              </div>
            </>
          )
      }
      </div>
    </>
  );
};

export default CountryList;
