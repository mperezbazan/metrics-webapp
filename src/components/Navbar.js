import { Link } from 'react-router-dom';

const Navbar = () => (
  <header className="header-container">
    <nav className="header-nav">
      <div className="header-list">
        <li className="header-item">
          <Link to="/">
            {'<'}
          </Link>
        </li>
        <li className="header-item">
          Country views
        </li>
        <li className="header-item">
          <div className="button-group">
            <i className="fa-solid fa-microphone" />
            <i className="fa-solid fa-gear" />
          </div>
        </li>
      </div>
    </nav>
  </header>
);

export default Navbar;
