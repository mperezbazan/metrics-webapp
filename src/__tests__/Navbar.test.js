import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';
import CountryList from '../components/CountryList';

describe('Test react components', () => {
  it('header renders', () => {
    const tree = renderer
      .create(
        <Router>
          <Navbar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Test CountryList renders', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <CountryList />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
