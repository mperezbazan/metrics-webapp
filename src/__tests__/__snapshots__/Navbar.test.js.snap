// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test react components Test CountryList renders 1`] = `
<div>
  <div
    className="loader-container"
  >
    <div
      className="dot-wave"
    >
      <div
        className="dot-wave__dot"
      />
      <div
        className="dot-wave__dot"
      />
      <div
        className="dot-wave__dot"
      />
      <div
        className="dot-wave__dot"
      />
    </div>
  </div>
</div>
`;

exports[`Test react components header renders 1`] = `
<header
  className="header-container"
>
  <nav
    className="header-nav"
  >
    <div
      className="header-list"
    >
      <li
        className="header-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          &lt;
        </a>
      </li>
      <li
        className="header-item"
      >
        Country views
      </li>
      <li
        className="header-item"
      >
        <div
          className="button-group"
        >
          <i
            className="fa-solid fa-microphone"
          />
          <i
            className="fa-solid fa-gear"
          />
        </div>
      </li>
    </div>
  </nav>
</header>
`;
